xbmc/utils/CryptThreading.cpp

```diff
@@ -66,14 +66,14 @@ static unsigned long thread_id()
 
 CryptThreadingInitializer::CryptThreadingInitializer()
 {
-  bool attemptedToSetSSLMTHook = false;
 #ifdef HAVE_OPENSSL
   // set up OpenSSL
   numlocks = CRYPTO_num_locks();
   CRYPTO_set_id_callback(thread_id);
   CRYPTO_set_locking_callback(lock_callback);
-  attemptedToSetSSLMTHook = true;
+  bool attemptedToSetSSLMTHook = true;
 #else
+  bool attemptedToSetSSLMTHook = false;
```

Comment:
Nope - was a similar but not the same situation. In the old code it was defined in a real else { } clause and went out of scope directly after it.

But I would just not do such optimizations. It does not harm if it's set to false out of that block and set to true in the #ifdef block